frontend/src/components/game/GamePage.jsx

```diff
@@ -0,0 +1,283 @@
+import { Status as TowerStatus } from "../../entities/towers/Tower";
+import { deleteSave, getUsername } from "../../utils/api_utils";
+import { GameProvider } from "./GameProvider";
+import GameContainer from "./GameContainer";
+import { useGameContext } from "./GameContext";
+
+const NAME = "GamePage";
+const BUY_KEY = "b";
+
+const HudTopDiv = styled('div')(({ theme }) => ({
+    ...theme.typography.button,
+    color: "white",
+    fontSize: "medium",
+}));
+
+const GamePage = ({ gameMap, devMode = true }) => {
+    const [game, setGame] = useState(() => new Game(new gameMap()));
+    const { keyboardInput } = game;
+
+    const [modal, setModal] = useState(devMode ? "" : "WELCOME");
+
+    const hideModal = () => setModal("");
+    const startMusicAndHideModal = () => {
+        Howler.volume(0.5);
+        const sound = new Howl({
+            src: ['audio/funkysuspense.mp3']
+        });
+        sound.play();
+
+        hideModal();
+    }
+
+    const deleteCurrentGame = async () => {
+        const token = localStorage.getItem('project-bk-token');
+        if (token) {
+            const decoded = jwtDecode(token);
+            const { userId } = decoded;
+            await deleteSave(userId, game.createdAt);
+        }
+    }
+    useEffect(() => {
+        game.addPhaseListener(BUILD, hideModal);
+        game.addPhaseListener(DEFEND, () => setModal(DEFEND)); // there is no modal for DEFEND, but this shows the phase in the HUD
+        game.addPhaseListener(SCORE, () => setModal(SCORE));
+        game.addPhaseListener(WIN, () => setModal(WIN));
+        game.addPhaseListener(LOSE, () => setModal(LOSE));
+        game.addPhaseListener(WIN, deleteCurrentGame);
+        game.addPhaseListener(LOSE, deleteCurrentGame);
+
+        keyboardInput.addKeyDownCallback(BUY_KEY, NAME, () => {
+            if (game.phase === BUILD) {
+                setModal("BUY");
+            }
+        })
+
+        return () => {
+            game.removeAllPhaseListeners(BUILD);
+            game.removeAllPhaseListeners(DEFEND);
+            game.removeAllPhaseListeners(SCORE);
+            game.removeAllPhaseListeners(WIN);
+            game.removeAllPhaseListeners(LOSE);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [game])
+
+    const [selectedTower, setSelectedTower] = useState(null);
+    const [errorMessage, setErrorMessage] = useState(null);
+
+    return (
+        <GameProvider game={game} >
+            {/* All Modals */}
+            <WelcomeModal
+                show={modal === "WELCOME"}
+                hideModal={startMusicAndHideModal}
+                setGame={setGame}
+            />
+            <ScorePhaseModal show={modal === SCORE} />
+            <WinModal onHide={hideModal} show={modal === WIN} />
+            <LoseModal onHide={hideModal} show={modal === LOSE} />
+            <BuyModal
+                open={modal === "BUY"}
+                setOpen={hideModal}
```

Comment:
Prolly worth renaming now that the function is different (doesn't take bool anymore)
```suggestion
                hideModal={hideModal}
```